c7n/resources/apigw.py

```diff
@@ -122,14 +124,29 @@ class RestApi(query.QueryResourceManager):
 
     class resource_type(query.TypeInfo):
         service = 'apigateway'
-        arn_type = 'restapis'
+        arn_type = '/restapis'
         enum_spec = ('get_rest_apis', 'items', None)
         id = 'id'
         name = 'name'
         date = 'createdDate'
         dimension = 'GatewayName'
         config_type = "AWS::ApiGateway::RestApi"
 
+    @property
+    def generate_arn(self):
+        """
+         Sample arn: arn:aws:apigateway:us-east-1::/restapis/rest-api-id
+         This method overrides c7n.utils.generate_arn and drops
+         account id from the generic arn.
+        """
+        if self._generate_arn is None:
+            self._generate_arn = functools.partial(
+                generate_arn,
+                self.resource_type.service,
+                region=not self.resource_type.global_resource and self.config.region or "",
```

Comment:
seems like region is always present so can just use self.config.region directly here

Reply:
Addressed, thanks!